code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/AEPPushTemplate.kt

```diff
@@ -0,0 +1,400 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.services.ui.notification
+
+import android.app.NotificationManager
+import android.content.Intent
+import android.os.Build
+import androidx.annotation.RequiresApi
+import androidx.core.app.NotificationCompat
+import com.adobe.marketing.mobile.services.Log
+import com.adobe.marketing.mobile.util.DataReader
+import com.adobe.marketing.mobile.util.StringUtils
+
+/**
+ * This class is used to parse the push template data payload or an intent and provide the necessary information
+ * to build a notification.
+ */
+internal sealed class AEPPushTemplate {
+    /** Enum to denote the type of action  */
+    enum class ActionType {
+        DEEPLINK, WEBURL, DISMISS, OPENAPP, NONE
+    }
+
+    /** Class representing the action button with label, link and type  */
+    class ActionButton(val label: String, val link: String?, type: String?) {
+        val type: ActionType
+
+        init {
+            this.type = ActionType.valueOf(type ?: ActionType.NONE.name)
+        }
+    }
+
+    internal object ActionButtonType {
+        const val DEEPLINK = "DEEPLINK"
+        const val WEBURL = "WEBURL"
+        const val DISMISS = "DISMISS"
+        const val OPENAPP = "OPENAPP"
+    }
+
+    internal object ActionButtons {
```

Comment:
How do feel about moving a few of the constants in this class like ActionButtons, NotificationPriority and NotificationVisibility to a constants file?

Reply:
we can definitely move them to the main PushConstants file